src/Network/Minio.hs

```diff
@@ -73,7 +73,11 @@ module Network.Minio
   , ObjectPartInfo(..)
   , listIncompleteParts
 
-  -- ** Notifications
+  -- ** Bucket Policies
```

Comment:
We can support policy operations at two levels. At a lower level, we can provide `getBucketPolicy` and `setBucketPolicy` as implemented in this PR and export it via `Network.Minio.S3API` module. In addition, we can provide higher level data types representing the policy JSON schema and APIs which allow users to construct policy JSON from the Haskell data type. This will help users to construct type-safe policies.

Reply:
This API is kept similar to what other libraries are doing, if we implement parsing logic then we need to handle dynamic JSON. It is not possible to do that cleanly - this is the reason we removed it in the first place. Generating policy is outside the scope of our libraries we are simply making it low level. 

I kept this code similar to BucketNotification API which also lists itself in S3API.hs